src/api/node/sse/types.rs

```diff
@@ -0,0 +1,49 @@
+use serde::{Deserialize, Serialize};
+
+//copied from casper-sidecar
+
+#[derive(Clone, Copy, Eq, PartialEq, Debug, Hash)]
+pub enum EventType {
+    ApiVersion,
+    SidecarVersion,
+    BlockAdded,
+    TransactionAccepted,
+    TransactionProcessed,
+    TransactionExpired,
+    Fault,
+    FinalitySignature,
+    Step,
+    Other,
+}
+
+// TODO: add full deserialization of the json
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Debug)]
+pub enum EventInfo {
+    ApiVersion(casper_types::ProtocolVersion),
+    SidecarVersion(serde_json::Value),
```

Comment:
I think supporting the additional `SidecarVersion` event in the SSE client shouldn't be an issue. Since the sidecar mirrors the SSE feed from the node with one extra event it makes sense to make it compatible with both. Especially with the uncertainty around sidecar adoption and whether the validators will prefer to run the SSE through sidecar or the node itself

Reply:
Well, additional variant is not the only change, and I expect the differences will grow with time :roll_eyes:. I would choose one source of truth.